lib/msal-browser/docs/accounts.md

```diff
@@ -35,25 +47,97 @@ function handleResponse(resp) {
-    request.account = myMSALObj.getAccountByHomeId(homeAccountId); // alternatively: myMSALObj.getAccountByLocalId(localAccountId) or myMSALObj.getAccountByUsername(username) 
+async function getTokenPopup(request, homeAccountId) {
+    // In this case, accounts are filtered by homeAccountId, but more attributes can be added to refine the search and increase the precision of the account filter
+    const accountFilter = {
+        homeAccountId: homeAccountId,
+    };
+    request.account = myMSALObj.getAccount(accountFilter);
     return await myMSALObj.acquireTokenSilent(request).catch(async (error) => {
-       // Handle error
+        // Handle error
         return await myMSALObj.acquireTokenPopup(request);
     });
 }
 ```
 
+### Filtering by login hint
+
+As of `@azure/msal-browser@3.2.0`, all login hint values can be used to search for and filter accounts. In order to filter by login hint, MSAL will compare the `loginHint` value in the `AccountFilter` object against the following account attributes (in order of precedence) to search for matches:
+
+-   `login_hint` ID token claim
+-   `username` account property
+-   `upn` ID token claim
+
+> Note: All attributes above can be passed into the account filter as the `loginHint` property. The account filter will also accept the `username` attribute as `username`, and will yield a more performant search.
+
+#### Using `login_hint` claim
+
+```javascript
+const accountFilter = {
+    loginHint: previouslyObtainedIdTokenClaims.login_hint;
+};
+request.account = myMSALObj.getAccount(accountFilter);
+return await myMSALObj.acquireTokenSilent(request).catch(async (error) => {
+    // Handle error
+    return await myMSALObj.acquireTokenPopup(request);
+});
+```
+
+#### Using `username``
+
+> Note: The `username` value can included in the `AccountFilter` object as either `username` or `loginHint`. This is because the `username` claim is one of the 3 values (along with the `login_hint` and `upn` ID token claims) that the token service accepts as login hint. If your application is certain that the value in question is a `username`, setting it as the `AccountFilter.username` property will yield better search performance. Being able to set a `username` value as `loginHint` is useful if your application utilizes a login hint and does not keep context on whether that value came from a `username`, `login_hint`, or `upn` claim.
+
+Passing `username` as `loginHint`
+
+```javascript
+const accountUsername = userProfile.username;
+const accountFilter = {
+    loginHint: accountUsername;
+};
+request.account = myMSALObj.getAccount(accountFilter);
+return await myMSALObj.acquireTokenSilent(request).catch(async (error) => {
+    // Handle error
+    return await myMSALObj.acquireTokenPopup(request);
+});
+```
+
+Passing `username` as `username`
+
+```javascript
+const accountUsername = userProfile.username;
+const accountFilter = {
+    username: accountUsername; // More performant filtering
```

Comment:
nit: The explanation above is sufficient, I don't want to inadvertently encourage folks to use username instead of the claim
```suggestion
    username: accountUsername;
```